containers/hawkular-metrics-index-truncator/src/main/java/org/hawkular/openshift/truncator/Truncator.java

```diff
@@ -0,0 +1,157 @@
+    private String heapsterRC;
+    private String keyspace;
+
+    private int partitionThreshold;
+    private int cassandraConnectionMaxRetries;
+
+    private int truncatorMaxRetries;
+    private long truncatorMaxDelay;
+    private long cassandraConnectionMaxDelay;
+
+    private OpenshiftClientWrapper openshiftClient;
+
+    private int metricsReplicas, heapsterReplicas;
+
+    public Truncator() {
+        KUBERNETES_MASTER_URL = System.getProperty(KUBERNETES_MASTER_URL_SYSPROP, KUBERNETES_MASTER_URL_DEFAULT);
+        metricsRC = System.getProperty("hawkular.metrics.rc-name", METRICS_RC_NAME_DEFAULT);
+        heapsterRC = System.getProperty("hawkular.metrics.heapster.rc-name", HEAPSTER_RC_NAME_DEFAULT);
+        cassandraPodPrefix = System.getProperty("hawkular.metrics.cassandra.pod-prefix", CASSANDRA_POD_PREFIX_DEFAULT);
+        partitionThreshold = Integer.getInteger("hawkular.metrics.partition-threshold", PARTITION_THRESHOLD_DEFAULT);
+        useSSL = Boolean.getBoolean("hawkular.metrics.cassandra.use-ssl");
+        forceTruncation = Boolean.getBoolean("hawkular.metrics.force-truncation");
+        keyspace = System.getProperty("hawkular.metrics.cassandra.keyspace", "hawkular_metrics");
+        namespace = System.getProperty("hawkular.metrics.namespace", "openshift-infra");
+        cassandraConnectionMaxDelay = Long.getLong("hawkular.metrics.cassandra.connection.max-delay", 30);
+        cassandraConnectionMaxRetries = Integer.getInteger("hawkular.metrics.cassandra.connection.max-retries", 5);
+        truncatorMaxRetries = Integer.getInteger("hawkular.metrics.truncator.max-retries", 5);
+        truncatorMaxDelay = Long.getLong("hawkular.metrics.truncator.max-retries", 5*1000);
+    }
+
+    private String getToken() throws IOException {
+        return new String(Files.readAllBytes(Paths.get(TOKEN_FILE_LOCATION)));
+    }
+
+    public void restoreControllers() {
+        if (this.heapsterReplicas != 0) {
+            openshiftClient.scaleReplicaitonController(this.heapsterRC, this.heapsterReplicas);
+        }
+
+        if (this.metricsReplicas != 0) {
+            openshiftClient.scaleReplicaitonController(this.metricsRC, this.metricsReplicas);
+        }
+    }
+
+    public void truncate() throws IOException, InterruptedException, RuntimeException {
+
+        metricsReplicas = heapsterReplicas = 0;
+        String token = this.getToken();
+        openshiftClient = new OpenshiftClientWrapper(this.namespace, this.KUBERNETES_MASTER_URL, token);
+
+        Pod[] cassandraPods = openshiftClient.getPodsWithPrefix(this.cassandraPodPrefix);
+        ArrayList<String> cassandraHosts = new ArrayList<>();
+        Arrays.stream(cassandraPods).forEach((pod) -> {
+            cassandraHosts.add(pod.getIP());
+        });
+
+        if (cassandraPods.length > 0 || this.forceTruncation) {
+
+            if (openshiftClient.hasLargePartition(this.partitionThreshold, this.keyspace, this.cassandraPodPrefix)) {
```

Comment:
shouldn't the `|| forceTruncation` check be here? I would understand it as "perform truncation even if there is no large partition", but that's not the case now

Reply:
Yes, nice catch!